### ISSUE TITLE
Rework tests

### ISSUE THREAD

**Issue by jajik** [modcluster/mod_proxy_cluster#112]
What that would be good to do:

- [ ] Make tests more modular
- [ ] Remove duplication
- [ ] Consider using local image repository instead of quay.io
- [ ] Run tests in CI (or a reasonable subset)

**Comment by jfclere** [comment]
I also fixed the following:

diff --git a/test/native/Dockerfile b/test/native/Dockerfile
index 3955f91..40b72bd 100644
--- a/test/native/Dockerfile
+++ b/test/native/Dockerfile
@@ -2,7 +2,7 @@ FROM tomcat:8.5
 
 WORKDIR /usr/local/tomcat
 
-COPY target/dependency/*.jar ./lib/
+COPY target/*.jar ./lib/
 COPY ./files/server.xml ./conf/
 COPY ./files/context.xml ./conf/
 COPY ./files/start.sh ./
diff --git a/test/native/pom.xml b/test/native/pom.xml
index 0c78ef8..88ee4b8 100644
--- a/test/native/pom.xml
+++ b/test/native/pom.xml
@@ -41,7 +41,7 @@
 -->
             <dependency>
                 <groupId>org.jboss.mod_cluster</groupId>
-                <artifactId>mod_cluster-container-tomcat85</artifactId>
+                <artifactId>mod_cluster-container-tomcat-8.5</artifactId>
                 <version>${project.version}</version>
             </dependency>
             <dependency>
diff --git a/test/native/tests.sh b/test/native/tests.sh
index a1f81b5..b543063 100644
--- a/test/native/tests.sh
+++ b/test/native/tests.sh
@@ -12,7 +12,7 @@ if [ -z ${ITERATION_COUNT+x} ]; then
     ITERATION_COUNT=50 # the number of iteration of starting/stopping a tomcat
 fi
 if [ -z ${IMG+x} ]; then
-    IMG=IMG=quay.io/$USER/tomcat_mod_cluster
+    IMG=quay.io/$USER/tomcat_mod_cluster
 fi
 
 echo -n "Test values are FOREVER_PAUSE=$FOREVER_PAUSE, TOMCAT_CYCLE_COUNT=$TOMCAT_CYCLE_COUNT,"

**Comment by jajik** [comment]
> I also fixed the following:
> 
> ```
> diff --git a/test/native/Dockerfile b/test/native/Dockerfile
> index 3955f91..40b72bd 100644
> --- a/test/native/Dockerfile
> +++ b/test/native/Dockerfile
> @@ -2,7 +2,7 @@ FROM tomcat:8.5
>  
>  WORKDIR /usr/local/tomcat
>  
> -COPY target/dependency/*.jar ./lib/
> +COPY target/*.jar ./lib/
>  COPY ./files/server.xml ./conf/
>  COPY ./files/context.xml ./conf/
>  COPY ./files/start.sh ./
> diff --git a/test/native/pom.xml b/test/native/pom.xml
> index 0c78ef8..88ee4b8 100644
> --- a/test/native/pom.xml
> +++ b/test/native/pom.xml
> @@ -41,7 +41,7 @@
>  -->
>              <dependency>
>                  <groupId>org.jboss.mod_cluster</groupId>
> -                <artifactId>mod_cluster-container-tomcat85</artifactId>
> +                <artifactId>mod_cluster-container-tomcat-8.5</artifactId>
>                  <version>${project.version}</version>
>              </dependency>
>              <dependency>
> diff --git a/test/native/tests.sh b/test/native/tests.sh
> index a1f81b5..b543063 100644
> --- a/test/native/tests.sh
> +++ b/test/native/tests.sh
> @@ -12,7 +12,7 @@ if [ -z ${ITERATION_COUNT+x} ]; then
>      ITERATION_COUNT=50 # the number of iteration of starting/stopping a tomcat
>  fi
>  if [ -z ${IMG+x} ]; then
> -    IMG=IMG=quay.io/$USER/tomcat_mod_cluster
> +    IMG=quay.io/$USER/tomcat_mod_cluster
>  fi
>  
>  echo -n "Test values are FOREVER_PAUSE=$FOREVER_PAUSE, TOMCAT_CYCLE_COUNT=$TOMCAT_CYCLE_COUNT,"
> ```

You're right, I forgot about it. I put it into #119 if you want to merge it. (I ran into issues while working on this; once I'll get CI working, I'll create another PR.)

**Comment by jajik** [comment]
Once #120 and #121 are merged, we can close this one.
